picamera2/encoders/h264_encoder.py

```diff
@@ -62,16 +63,21 @@ def _start(self):
         fmt.fmt.pix_mp.plane_fmt[0].sizeimage = 512 << 10
         fcntl.ioctl(self.vd, VIDIOC_S_FMT, fmt)
 
+        controls = [(V4L2_CID_MPEG_VIDEO_GOP_SIZE, self._gopsize)]
```

Comment:
I have a sneaky suspicion this should be V4L2_CID_MPEG_VIDEO_H264_I_PERIOD. What a fabulous interface, it's as though someone designed it to trip people up! That control parameter probably wants adding to our v4l2.py, I tried it and it seemed like it wasn't there.

Reply:
Thanks, that did the trick :) I've just added that flag to the v4l2 package and pushed to pypi too.

Do you think I should rename gopsize to iframeperiod possibly?

I ran v4l2-ctl --all --device /dev/video11

And it seems to be listed as i frame period in there too.

```
            h264_i_frame_period 0x00990a66 (int)    : min=0 max=2147483647 step=1 default=60 value=60
```